secretsauce/apps/account/views.py

```diff
@@ -33,26 +33,15 @@ class InviteUser(generics.CreateAPIView):
     permission_classes = [permissions.IsAdminUser]
 
     def create(self, request):
-        password = generatePassword()
-        user_email = request.data['email']
-        company_id = request.data['company']
-
-        company_instance = Company.objects.get(pk=company_id)
-
-        user = User.objects.create_user(
-            user_email,
-            password,
-            company = company_instance
-        )
-        user.save()
-        serializer=UserCreateSerializer(user)
-
-        mappings = {
-            'email': user_email,
-            'password':  password,
-        }
-
-        send_email('You have been invited to RMS Pricing Analytics Platform!', 'donotreply@rmsportal.com', [user_email], '', 'create_user.html', mappings)
-
-
-        return Response(serializer.data, status=status.HTTP_201_CREATED)
+        request.data['password'] = generatePassword()
+        serializer = UserCreateSerializer(data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            mappings = {
+            'email': serializer.data.get('email'),
+            'password':  serializer.data.get('password'),
+            }
+            send_email('You have been invited to RMS Pricing Analytics Platform!', 'donotreply@rmsportal.com', [mappings['email']], '', 'create_user.html', mappings)
+
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)        
```

Comment:
Did you test the endpoint directly? Can't get it to work.

Seems like the request.data dictionary cant be modified? Can clone `request.data` and pass it into the serializer.
Also, frontend will be passing the company_id over. I am not sure whether the drf serializer will be able to match the company_id to the company instance
![image](https://user-images.githubusercontent.com/39354695/86211287-ba235900-bba8-11ea-933d-f2ac0285dcb8.png)

Reply:
Also found the UserCreateSerializer source code if u are interested
https://github.com/sunscrapers/djoser/blob/6ed0a05522234e56b3c9b9ad06fc08079a73984c/djoser/serializers.py

Reply:
Fixed the error, should be able to work now.
Serializer matches the company.id since its the primary key.

Reply:
`serializer.data.get('password')` does not return the password as i think the serializer does not return the password. Fixed the bug alr. Good to go.